File: GEMA_andorid/GEMAINVENTORY/app/src/main/java/com/example/gemainventory/ui/settings/SettingsManageFragment.java
package com.example.gemainventory.ui.settings;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.example.gemainventory.R;

public class SettingsManageFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_settings_manage, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // 1. Botón Atrás (Fundamental)
        view.findViewById(R.id.btn_back).setOnClickListener(v -> {
            NavHostFragment.findNavController(this).popBackStack();
        });

        // 2. Gestionar Productos
        view.findViewById(R.id.btn_manage_products).setOnClickListener(v -> {
            // Opción A: Redirigir al Inventario existente (Recomendado)
            NavHostFragment.findNavController(this).navigate(R.id.navigation_inventory);
        });

        // 3. Gestionar Categorías (Placeholder)
        view.findViewById(R.id.btn_manage_categories).setOnClickListener(v -> {
            NavHostFragment.findNavController(this)
                    .navigate(R.id.action_manage_to_categories); // <--- ID que creamos arriba
        });

        // 4. Gestionar Clientes (Placeholder)
        view.findViewById(R.id.btn_manage_clients).setOnClickListener(v -> {
            NavHostFragment.findNavController(this)
                    .navigate(R.id.action_manage_to_clients); // <--- Usa el ID que pusiste en el nav graph
        });

        // 5. Gestionar Almacenes (Placeholder)
        view.findViewById(R.id.btn_manage_warehouses).setOnClickListener(v -> {
            NavHostFragment.findNavController(this).navigate(R.id.action_manage_to_warehouses);
        });

        // 6. Gestionar Proveedores
        view.findViewById(R.id.btn_manage_suppliers).setOnClickListener(v -> {
            NavHostFragment.findNavController(this).navigate(R.id.action_manage_to_suppliers);
        });
    }
}

File: GEMA_andorid/GEMAINVENTORY/app/src/main/java/com/example/gemainventory/ui/settings/SettingsManageFragment.kt
package com.example.gemainventory.ui.settings

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.example.gemainventory.R

class SettingsManageFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val prefs = requireContext().getSharedPreferences("GemaPrefs", Context.MODE_PRIVATE)
        val isDarkMode = prefs.getBoolean("DarkMode", false)

        return ComposeView(requireContext()).apply {
            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
            setContent {
                SettingsManageScreen(
                    darkTheme = isDarkMode,
                    onBackClick = {
                        findNavController().popBackStack()
                    },
                    onManageProducts = {
                        findNavController().navigate(R.id.navigation_inventory)
                    },
                    onManageCategories = {
                        findNavController().navigate(R.id.action_manage_to_categories)
                    },
                    onManageClients = {
                        findNavController().navigate(R.id.action_manage_to_clients)
                    },
                    onManageWarehouses = {
                        findNavController().navigate(R.id.action_manage_to_warehouses)
                    },
                    onManageSuppliers = {
                        findNavController().navigate(R.id.action_manage_to_suppliers)
                    }
                )
            }
        }
    }
}
